### PR TITLE
Countervalues Rework

### PR DESCRIPTION
**Pull request by gre**
first iteration of a new PoC of the countervalues logic. It is not yet ready for production as iteration is ongoing.

### CODE REVIEW

File: src/countervalues/react.js [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+  );
+
+  // trigger poll but not every time poll callback is changed
+  const [triggerPoll, setTriggerPoll] = useState(false);
+
+  useEffect(() => {
+    if (pending || !triggerPoll) return;
+    dispatch({ type: "pending" });
+    loadCountervalues(state, userSettings).then(
+      (state) => {
```
Comment (gre) [inline src/countervalues/react.js:104]: ⚠️  there are race condition here if a second effect runs before the end of the first

File: src/countervalues/react.js [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+  }, [autopollInterval, pollInitDelay, isPolling]);
+
+  const polling = useMemo<Polling>(
+    () => ({
+      wipe: () => {
+        dispatch({ type: "wipe" });
+      },
+      poll: () => setTriggerPoll(true),
+      start: () => setIsPolling(true),
+      stop: () => setIsPolling(false),
```
Comment (gre) [inline src/countervalues/react.js:162]: I don't understand the distinction of the `isPolling` and `pending`

File: src/countervalues/react.js [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+      if (
+        JSON.stringify(userSettings) === JSON.stringify(userSettingsRef.current)
+      )
+        return;
+    }
+    setTriggerPoll(true);
+    userSettingsRef.current = userSettings;
+  }, [userSettings]);
+
+  const [isPolling, setIsPolling] = useState(true);
```
Comment (gre) [inline src/countervalues/react.js:141]: I don't understand the difference between `isPolling` and `pending`

Comment (JunichiSugiura) [reply src/countervalues/react.js:141]: `isPolling` state is used to trigger polling effect where `pending` is the state.
https://github.com/LedgerHQ/ledger-live-common/blob/countervalues-poc/src/countervalues/react.js#L160-L161

Comment (JunichiSugiura) [reply src/countervalues/react.js:141]: `triggerPollling` would be more appropriate name

Comment (gre) [reply src/countervalues/react.js:141]: but there is already a triggerPoll 🤔

Comment (JunichiSugiura) [reply src/countervalues/react.js:141]: oh, i've noticed that `triggerPolling` also exists.

Comment (JunichiSugiura) [reply src/countervalues/react.js:141]: `triggerPolling` for one shot fetching and `isPolling` for continuous syncing.

**Pull request closed without merging**